java/client/src/main/java/glide/connectors/handlers/PubSubMessageQueue.java

```diff
@@ -0,0 +1,82 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.connectors.handlers;
+
+import glide.api.models.PubSubMessage;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.Future;
+
+/** A FIFO message queue for {@link PubSubMessage} backed by {@link ConcurrentLinkedDeque}. */
+public class PubSubMessageQueue {
+    /** The queue itself. */
+    private final ConcurrentLinkedDeque<PubSubMessage> mq = new ConcurrentLinkedDeque<>();
```

Comment:
Do you actually need this to be a Concurrent* object if all method calls on it are in synchronized blocks already? Let's expand mq to messageQueue for better readability too.